_rules/text-node-contrast-afw4f7.md

```diff
@@ -0,0 +1,337 @@
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+Black [text node](https://dom.spec.whatwg.org/#text) on a white background.
+
+```html
+<p style="color: #333; background: #FFF;">
+	Some text in English
+</p>
+```
+
+#### Passed Example 2
+
+Black [text node](https://dom.spec.whatwg.org/#text) on a partially white gradient background.
+
+```html
+<p style="color: #333; background: linear-gradient(to right, #FFF, #00F);">
+	Some text in English
+</p>
+```
+
+#### Passed Example 3
+
+Light [text node](https://dom.spec.whatwg.org/#text) on a partially black background image.
+
+```html
+<p
+	style="color: #CCC; height:50px; padding-top:25px; background: #000 no-repeat -20px -20px url('../test-assets/black-hole.jpeg');"
+>
+	Black hole sun
+</p>
+```
+
+#### Passed Example 4
+
+[text node](https://dom.spec.whatwg.org/#text) that passes because of a text shadow.
+
+```html
+<p style="color: #000; background: #737373; text-shadow: white 0 0 3px">
+	Some text in English
+</p>
+```
+
+#### Passed Example 5
+
+18 point text with a 3:1 contrast.
+
+```html
+<p style="color: #000; font-size:18pt; background: #666;">
+	Some text in English
+</p>
+```
+
+#### Passed Example 6
+
+14 point bold text with a 3:1 contrast.
+
+```html
+<p style="color: #000; font-size:14pt; font-weight:700; background: #666;">
+	Some text in English
+</p>
+```
+
+#### Passed Example 7
+
+Decorative text with insufficient contrast.
+
+```html
+<p>Helvetica is a widely used sans-serif typeface developed in 1957 by Max Miedinger and Eduard Hoffmann.</p>
+<p style="font-family: helvetica; background: #EEE; color: #777;" aria-hidden="true">
+	The quick brown fox jumps over the lazy dog.
+</p>
+```
+
+#### Passed Example 8
+
+Text that does not express anything in human language.
+
+```html
+<p style="color: #000; background: #666;">
+	----=====++++++++___________***********%%%%%%%%%%%±±±±@@@@@@@@
+</p>
+```
+
+#### Passed Example 9
+
+Text that is in a shadow DOM tree with sufficient contrast.
+
+```html
+<p style="color: #333; background: #fff;" id="p"></p>
+<script>
+  const shadowRoot = document.getElementById('p').attachShadow({ mode: 'open' })
+  shadowRoot.textContent = 'Some text in English'
+</script>
+```
+
+### Failed
+
+#### Failed Example 1
+
+Text with insufficient contrast on a plain background.
+
+```html
+<p style="color: #AAA; background: white;">
+	Some text in English
+</p>
+```
+
+#### Failed Example 2
+
+Text with insufficient contrast on a gradient background.
+
+```html
+<p style="color: #AAA; background: linear-gradient(to right, #FFF, #00F);">
+	Some text in English
+</p>
+```
+
+#### Failed Example 3
+
+Text with insufficient contrast on a background image.
+
+```html
+<p
+	style="color: #555; height:50px; padding-top:25px; background: black no-repeat -20px -20px url('../test-assets/black-hole.jpeg');"
```

Comment:
can we add the test-assets under a specific directory pertaining to the ruleId?

eg: `test-assets/XXXXXX/something.xyz`